src/plugins/controls/server/options_list/options_list_queries.ts

```diff
@@ -139,6 +142,74 @@ const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } =
       ),
   },
 
+  /**
+   * the "IP" query / parser should be used when the options list is built on a field of type IP.
+   */
+  ip: {
+    buildAggregation: ({ fieldName, searchString }: OptionsListRequestBody) => {
+      const ipSuggestions = {
+        terms: {
+          field: fieldName,
+          execution_hint: 'map',
+          shard_size: 10,
+        },
+      };
+
+      if (!searchString) {
+        return ipSuggestions;
+      }
+
+      const ipSegments = searchString
+        .replace(/[^\d.]/g, '') // remove any non-numeric characters, excluding periods
+        .split('.')
+        .filter((segment) => segment !== '');
+
+      // if a search string is provided that already is of the form `a.b.c.d` then simply search for that IP directly
+      if (ipSegments.length === 4) {
+        return {
+          filter: {
+            term: { [fieldName]: searchString },
+          },
+          aggs: {
+            filteredSuggestions: { ...ipSuggestions },
+          },
+        };
+      }
+
+      // otherwise, find IPs that **start** with the given partial IP - for example, if the search string is `a.b`
+      // then do a search for IPs in the range `a.b.0.0` to `a.b.255.255`
+      const minIp = ipSegments.concat(Array(4 - ipSegments.length).fill('0')).join('.');
+      const maxIp = ipSegments.concat(Array(4 - ipSegments.length).fill('255')).join('.');
+      return {
+        ip_range: {
+          field: fieldName,
+          ranges: [{ key: 'results', from: minIp, to: maxIp }],
+        },
+        aggs: {
+          filteredSuggestions: { ...ipSuggestions },
+        },
+      };
+    },
+    parse: (rawEsResult) => {
+      const buckets: [{ key: string; filteredSuggestions?: { buckets: [{ key: string }] } }] = get(
+        rawEsResult,
+        'aggregations.suggestions.buckets'
+      );
+      if (!Boolean(buckets)) {
+        // this means that a full IP was provided as a search string
+        return get(rawEsResult, 'aggregations.suggestions.filteredSuggestions.buckets')?.map(
+          (suggestion: { key: string }) => suggestion.key
+        );
+      }
+      if (buckets[0].filteredSuggestions) {
+        // this means that a partial IP was provided as a search string
+        return buckets[0].filteredSuggestions.buckets.map((suggestion) => suggestion.key);
+      }
+      // this means that no search string has been provided
+      return buckets.map((suggestion) => suggestion.key);
```

Comment:
Once again, kind of ugly code. Don't think there is a better way of doing this though, since the Elasticsearch results vary in structure depending on the three possible search string cases defined in `buildAggregation` (I.e. no search string, partial IP search string, or full IP search string)